services/src/system_cache.cpp

```diff
@@ -40,8 +41,8 @@ int SystemCache::get(SystemCacheKey key, void* value, size_t length) {
 }
 
 int SystemCache::set(SystemCacheKey key, const void* value, size_t length) {
-    CHECK_TRUE(length <= sizeof(uint16_t), SYSTEM_ERROR_TOO_LARGE);
-    return tlv_.set(to_underlying(key), (const uint8_t*)value, length);
+    CHECK_TRUE(length <= std::numeric_limits<uint16_t>::max(), SYSTEM_ERROR_TOO_LARGE);
+    return tlv_.set(to_underlying(key), (const uint8_t*)value, length, 0);
```

Comment:
Is using `index == 0` here instead of the default -1 an optimization to avoid going through the entire file when deleting the entry?

Reply:
Correct.